### PR TITLE
Show unused imports as editor suggestion

### PR DESCRIPTION
**Pull request by joeandaverde**
**Extension Changes**

- Show unused imports (can be disabled via settings)
- Fixed code action for removing unused import.
- Improve definition provider to consider many modules using the same alias.
- Workspace symbol caching 
- Use https://github.com/joeandaverde/elm-project-inspect to begin consolidating elm project related tasks. Presently used for module resolution.

**Development changes**

- Update vscode engine
- Fix debug launch task
- Add prettierrc
- Upgrade typescript

### CODE REVIEW

File: src/elmUnusedImports.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        ];
+      };
+
+      return getImportsAllDiagnostics();
+    }
+
+    return [];
+  }));
+
+  return unusedImportDiagnostics.reduce((acc, x) => acc.concat(x), []);
```
Comment (joeandaverde) [inline src/elmUnusedImports.ts]: This is done quite a bit. Any problem with pulling in `lodash`?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "trailingComma": "all",
+  "tabWidth": 2,
+  "semi": true,
+  "singleQuote": true,
+  "printWidth": 80,
```
Comment (joeandaverde) [inline .prettierrc]: In my opinion, this is far too narrow. I'd like to propose a larger limit such as 120.

File: src/elmProjectManager.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import * as vscode from 'vscode';
+import { ElmProjectManager } from 'elm-project-inspect';
+import * as _ from 'lodash';
+
+export const getGlobalProjectManager = _.memoize(() => {
+  return new ElmProjectManager(
+    (vscode.workspace.workspaceFolders || []).map(x => x.uri.fsPath),
```
Comment (joeandaverde) [inline src/elmProjectManager.ts]: Haven't experimented with this much. There may be an event to indicate when the workspace folders change.

File: .prettierrc (hunk 2 above)
Comment (Krzysztof-Cieslak) [reply .prettierrc]: 120 sounds good to me

**Pull request closed without merging**